src/duty/sync/sync.service.ts

```diff
@@ -22,44 +22,78 @@ export class SyncService {
+    this.logger.log(`Getting sync committee participation info for state slot ${stateSlot}`);
+    const syncCommitteeBits = new BitVectorType(SYNC_COMMITTEE_SIZE); // sync participants count in committee
     const indexedValidators = await this.getSyncCommitteeIndexedValidators(epoch, stateSlot);
     this.logger.log(`Processing sync committee participation info`);
     const epochBlocks: BlockInfoResponse[] = [];
     const missedSlots: number[] = [];
     const startSlot = epoch * this.config.get('FETCH_INTERVAL_SLOTS');
     for (let slot = startSlot; slot < startSlot + this.config.get('FETCH_INTERVAL_SLOTS'); slot = slot + 1) {
       const blockInfo = await this.clClient.getBlockInfo(slot);
-      blockInfo ? epochBlocks.push(blockInfo) : missedSlots.push(slot);
+      if (blockInfo) {
+        epochBlocks.push(blockInfo);
+      } else {
+        missedSlots.push(slot);
+      }
     }
+
     this.logger.debug(`All missed slots in getting sync committee info process: ${missedSlots}`);
+    /**
+     * @todo is it possible to have all missed slots in an epoch (and so, zero epoch blocks)?
+     */
     const epochBlocksBits = epochBlocks.map((block) => {
       return {
         block: Number(block.message.slot),
-        bits: SyncCommitteeBits.deserialize(fromHexString(block.message.body.sync_aggregate.sync_committee_bits)),
+        bits: syncCommitteeBits.deserialize(fromHexString(block.message.body.sync_aggregate.sync_committee_bits)),
       };
     });
+
     for (const indexedValidator of indexedValidators) {
-      const synced_blocks: number[] = [];
+      const syncedBlocks: number[] = [];
       for (const blockBits of epochBlocksBits) {
         if (blockBits.bits.get(indexedValidator.in_committee_index)) {
-          synced_blocks.push(blockBits.block);
+          syncedBlocks.push(blockBits.block);
         }
       }
+
       const index = Number(indexedValidator.validator_index);
-      const percent = (synced_blocks.length / epochBlocksBits.length) * 100;
-      this.summary.epoch(epoch).set({
-        epoch,
-        val_id: index,
-        is_sync: true,
-        sync_percent: percent,
-        sync_meta: {
-          synced_blocks,
-        },
-      });
+      const summaryValidator = this.summary.epoch(epoch).get(index);
+
+      if (summaryValidator?.is_sync) {
+        let percentSum = 0;
+        for (const syncMetaItem of summaryValidator.sync_meta) {
+          percentSum += (syncMetaItem.synced_blocks.length / epochBlocksBits.length) * 100;
+        }
+        percentSum += (syncedBlocks.length / epochBlocksBits.length) * 100;
+
+        const newSyncMeta = [...summaryValidator.sync_meta];
+        newSyncMeta.push({
+          synced_blocks: syncedBlocks,
+        });
+
+        this.summary.epoch(epoch).set({
+          epoch,
+          val_id: index,
+          sync_percent: percentSum / (summaryValidator.sync_meta.length + 1),
+          sync_meta: newSyncMeta,
+        });
+      } else {
```

Comment:
I think this part might be simplified. just add one more object to `sync_meta` and have the same `sync_percent`